week-3.md

```diff
@@ -0,0 +1,63 @@
+
+## Schnorr
+
+The schnorr BIP was the first of the three BIPs with a [public draft](https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2018-July/016203.html), and is the one most easily analysed on its own, separate from the other BIPs. However, this is also, in some sense, the BIP that is least connected to Bitcoin -- it just describes a signature system, and leaves how it might be used in practice undocumented.
+
+Fortunately, having hopefully read the taproot and tapscript BIPs already, you have some ideas on how it might be used in Bitcoin! But you might like to also consider how the schnorr BIP might be useful elsewhere as well -- for example in the Bitcoin p2p protocol, for authenticating lightning gossip, for a new variant of Bitcoin signed messages (eg, [BIP 322](https://github.com/bitcoin/bips/blob/master/bip-0322.mediawiki)), or some other application.
+
+Anyway, the BIP is at:
+
+ * https://github.com/sipa/bips/blob/bip-schnorr/bip-schnorr.mediawiki
+
+You may wish to skim the "Applications" section since we've got week 5 reserved for looking into that in more detail. However most of the complexity there is actually in reading the detailed explanations of how those work in other papers and documents, so just reading the whole BIP this week might be a good approach too.
+
+Jonas Nick wrote a [blog post](https://medium.com/blockstream/reducing-bitcoin-transaction-sizes-with-x-only-pubkeys-f86476af05d7) for the Blockstream blog about x-only pubkey construction that might be interesting. t-bast from group 17 pointed out a [paper on bad nonces with ECDSA](https://eprint.iacr.org/2019/023.pdf) that may be useful background for why deterministic nonce generation is useful when possible; and Leonid Beder from the same group pointed to a [talk by Andrew Poelstra at the Monero Conference on Schnorr signatures](https://www.youtube.com/watch?v=L6KqkrP_nU4) that mentions an interesting construction to ensure proper nonce generation by hardware wallets. The videos from the [optech taproot workshop](https://bitcoinops.org/en/schorr-taproot-workshop/) might be helpful as a slightly gentler introduction to the topic.
+
+If you're interested in looking at implementations, then you may wish to look at any or all of:
+
+ * the [python reference code](https://github.com/sipa/bips/tree/bip-schnorr/bip-schnorr/) (not for production use)
+ * the [jupyter notebooks from the optech taproot workshop](https://github.com/bitcoinops/taproot-workshop/) (also not for production use)
+ * [libsecp256k1 PR#558](https://github.com/bitcoin-core/secp256k1/pull/558) adds basic support schnorr signature
```

Comment:
s/support/support for/ ?
s/signature/signatures/ ?

Reply:
changed